browser/ai_chat/ai_chat_settings_helper.cc

```diff
@@ -45,34 +38,79 @@ AIChatMojomHelperAndroid::AIChatMojomHelperAndroid(
       skus_service_getter, g_browser_process->local_state());
 }
 
-AIChatMojomHelperAndroid::~AIChatMojomHelperAndroid() = default;
-
-void AIChatMojomHelperAndroid::Destroy(JNIEnv* env) {
-  delete this;
-}
-
-jlong AIChatMojomHelperAndroid::GetInterfaceToAndroidHelper(JNIEnv* env) {
-  mojo::PendingRemote<mojom::AIChatAndroidHelper> remote;
-  receivers_.Add(this, remote.InitWithNewPipeAndPassReceiver());
+AIChatSettingsHelper::~AIChatSettingsHelper() = default;
 
-  return static_cast<jlong>(remote.PassPipe().release().value());
-}
-
-void AIChatMojomHelperAndroid::GetPremiumStatus(
-    GetPremiumStatusCallback callback) {
+void AIChatSettingsHelper::GetPremiumStatus(GetPremiumStatusCallback callback) {
   credential_manager_->GetPremiumStatus(
-      base::BindOnce(&AIChatMojomHelperAndroid::OnPremiumStatusReceived,
+      base::BindOnce(&AIChatSettingsHelper::OnPremiumStatusReceived,
                      weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
 }
 
-void AIChatMojomHelperAndroid::OnPremiumStatusReceived(
+void AIChatSettingsHelper::OnPremiumStatusReceived(
     mojom::PageHandler::GetPremiumStatusCallback parent_callback,
     mojom::PremiumStatus premium_status,
     mojom::PremiumInfoPtr premium_info) {
   std::move(parent_callback).Run(premium_status, std::move(premium_info));
 }
 
-void AIChatMojomHelperAndroid::CreateOrderId(CreateOrderIdCallback callback) {
+void AIChatSettingsHelper::GetModelsWithSubtitles(
+    GetModelsWithSubtitlesCallback callback) {
+  auto all_models = GetAllModels();
+  std::vector<mojom::ModelWithSubtitlePtr> models(all_models.size());
+  for (size_t i = 0; i < all_models.size(); i++) {
+    mojom::ModelWithSubtitle modelWithSubtitle;
+    modelWithSubtitle.model = all_models[i].Clone();
+    if (modelWithSubtitle.model->key == "chat-basic") {
+      modelWithSubtitle.subtitle =
+          l10n_util::GetStringUTF8(IDS_CHAT_UI_CHAT_BASIC_SUBTITLE);
+    } else if (modelWithSubtitle.model->key == "chat-leo-expanded") {
+      modelWithSubtitle.subtitle =
+          l10n_util::GetStringUTF8(IDS_CHAT_UI_CHAT_LEO_EXPANDED_SUBTITLE);
+    } else if (modelWithSubtitle.model->key == "chat-claude-instant") {
+      modelWithSubtitle.subtitle =
+          l10n_util::GetStringUTF8(IDS_CHAT_UI_CHAT_CLAUDE_INSTANT_SUBTITLE);
+    }
+    models[i] = modelWithSubtitle.Clone();
+  }
```

Comment:
Maybe put a DCHECK here for if it's not a key this function knows about so that we can remember to add the case when we add a new model. A test to catch that would probably be best though

Reply:
added dcheck